### PR TITLE
Update ginkgo to v2.12.0

### PR DESCRIPTION
**Pull request by saschagrunert**
#### What type of PR is this?

/kind ci

#### What this PR does / why we need it:
Update the ginkgo version to match the latest one in CI. 
#### Which issue(s) this PR fixes:

None
#### Special notes for your reviewer:
None
#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -56,7 +56,7 @@ integration-static: .install.ginkgo # It needs to be release so we correctly tes
 		$(MAKE) release-static; \
 	fi && \
 	export RUNTIME_BINARY="$(RUNTIME_PATH)" && \
-	export MAX_RSS_KB=5000 && \
+	export MAX_RSS_KB=7000 && \
```
Comment (saschagrunert) [inline Makefile:59]: Hm, I'm not exactly sure which recent bump caused this :thinking:

**Comment by rphillips** [comment]
@saschagrunert could you rebase this with main - pulling in the memory patch?

**Pull request merged**